src/utilities/utilities.ts

```diff
@@ -158,17 +158,22 @@ export async function execSwift(
  * @param args original commandline arguments
  */
 export function withSwiftSDKFlags(args: string[]): string[] {
-    switch (args.length > 0 ? args[0] : null) {
-        case "package": {
-            // swift-package requires SDK flags to be placed before subcommand options
-            // eg. ["package", "describe", "--type", "json"] should be turned into
-            // ["package", "describe", "--sdk", "/path/to/sdk", "--type", "json"]
-            if (args.length <= 2) {
-                return args.concat(swiftpmSDKFlags());
+    switch (args[0]) {
+        case "package":
+            switch (args[1]) {
+                case "dump-symbol-graph":
+                case "diagnose-api-breaking-changes": {
```

Comment:
Is there any harm in including the --sdk command line parameter even when it isn't needed. What's to say another package command requires the sdk.

Reply:
I would suggest to add them explicitly (since they're part of SwiftPM release). Adding unnecessary SDK flags has no effect I guess, but it will make the command line long and confusing (especially when the destination flags adds up).